bokehjs/src/lib/models/widgets/tables/data_table.ts

```diff
@@ -247,19 +265,51 @@ export class DataTableView extends WidgetView {
       reorderable = false
     }
 
+    let frozenRow = -1
+    let frozenBottom = false
+    const frozenColumn = this.model.frozen_columns == null ? -1 : this.model.frozen_columns-1
+    if (this.model.frozen_rows != null) {
+      frozenBottom = this.model.frozen_rows < 0 ? true : false
+      frozenRow = Math.abs(this.model.frozen_rows)
+    }
+
+    let autosize: string
+    if (this.model.fit_columns === true)
+      autosize = AutosizeModes.force_fit
+    else if (this.model.fit_columns === false)
+      autosize = AutosizeModes.off
+    else
+      autosize = (AutosizeModes as any)[this.model.autosize_mode]
```

Comment:
This should be factored out into a shared method between DataTable and DataCube.